theano/tensor/tests/test_basic.py

```diff
@@ -3088,6 +3091,20 @@ def test2(self):
             v_shape = eval_outputs(max_and_argmax(n, axis)[0].shape)
             assert tuple(v_shape) == np.max(data, np_axis).shape
 
+    def test2_float16(self):
+        data = rand(2, 3).astype("float16")
```

Comment:
We should at least test for negative numbers, and a couple of large ones, so make sure we are not just taking the argmax of the int16 representation of the float16 number.